static/js/find-a-partner.js

```diff
@@ -0,0 +1,170 @@
+(function() {
+
+  var filters = [];
+
+  function init() {
+    revealSearch();
+
+    var inputNode = document.querySelector(".p-find-a-partner__search-input");
+    inputNode.addEventListener("keyup", searchHandler);
+
+    var checkboxes = document.querySelectorAll(".p-find-a-partner__filter");
+    checkboxes.forEach(function(checkbox) {
+      checkbox.addEventListener("change", filter);
+    });
+
+    populateCheckboxes();
+    populateTextbox();
+  }
+
+  function matchesExist() {
+    var numberOfPartners = document.querySelectorAll(
+      ".p-find-a-partner__partner"
+    ).length;
+    var numberOfHidden = document.querySelectorAll(
+      ".p-find-a-partner__partner.u-hide"
+    ).length;
+    if (numberOfHidden === numberOfPartners) {
```

Comment:
You could just check if any are hidden. In doing so, removing the `numberOfPartners` query and just checking if `numberOfHidden` is greater then 0.